### DISCUSSION TITLE
Clunky drag and rotate

### DISCUSSION THREAD

**Discussion by zRelux** [pmndrs/use-gesture#250] [Q&A] [answered]
How do make a div that by the end of it I can drag and rotate it? I will paste a link of what I have achieved so far and technically it does the thing but it is super clunky and doesn't work very well... am I doing something wrong?

**Sandbox or Video**
https://codesandbox.io/s/admiring-rain-1d4yp

**Checklist:**
  - [x] I've read the [documentation](https://use-gesture.netlify.app/).
  - [x] If this is an issue with drag, I've tried setting `touch-action: none` to the draggable element.

**Comment by dbismut** [comment, marked as answer]
I'm sorry to say but you're using this lib the wrong way 😅 In this case you should only use the drag gesture. The move gesture should only be used when hovering an element.

Here is a sandbox where I'm giving away some tricks on how to handle what I guess you want to do (use the yellow shapes):
https://codesandbox.io/s/suspicious-heisenberg-04e8v

I've had a look at your `Ruler` file and fixed it a bit to allow for resizing from the red handle.

But you can also have a look at the `Alt` (for alternative solution) file that uses svg to avoid trigonometry calculations. You should rather look at the logic rather than the actual implementation if you need to continue using divs.

(if you want to test your original component you should probably comment out mine).

**Reply by zRelux** [reply]
I appreciate taking the time to help me with this, I'm still new to both libraries hence this mess... 🤣🤣
And I want to apologies again for making the issue again but I didn't see that you moved it to discussion.

The Alt component that you put looks amazing but I have two questions about it.

1. Is it possible to make it move more frequently (like more in real time)?
2. I would also like to add a long press detection on the circles at the end of the div that would zoom an image in the background, do you think it is possible?

**Reply by dbismut** [reply]
I’m not sure I understand point 2. But regarding point 1 you should look at the `immediate: true` option from react-spring which makes the animation 1:1. 

You’re not trying to pinch and zoom, are you? As this is natively supported in react-use-gesture. 

(No worries about the duplicate issue, GitHub notifications are a bit random).

**Reply by zRelux** [reply]
Thanks for tip 1. About tip 2 imagine a image right behind the ruler, basically if the user long presses the dots the image in the background would zoom to that point

**Reply by dbismut** [reply]
Still unsure about point 2. What do you mean by "zoom to that point"? In any case long press can easily be emulated with useDrag.

**Reply by zRelux** [reply]
How do you do that? can you provide a snippet for the longpress?

**Reply by dbismut** [reply]
One would be to set a timer with setTimeout when the user starts the drag and clear it when the user moves or release the pointer.

**Reply by zRelux** [reply]
Sorry would you mind sharing a quick snippet for this?

**Reply by dbismut** [reply]
There https://codesandbox.io/s/fervent-hellman-ml9f7
